src/lammpsparser/potential.py

```
@@ -33,6 +33,383 @@
+        """
+        # Normalize to set
+        if isinstance(target_elements, str):
+            self.target_elements = {target_elements}
+        elif isinstance(target_elements, (list, tuple)):
+            self.target_elements = set(target_elements)
+        elif isinstance(target_elements, set):
+            self.target_elements = target_elements
+        else:
+            raise ValueError(
+                f"target_elements must be str, list, or set, got {type(target_elements)}"
+            )
```

Ruff (0.15.0)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)